light_curves/code_src/panstarrs_functions.py

```diff
@@ -1,236 +1,16 @@
-    """Return metadata for the specified catalog and table
-    
-    Parameters
-    ----------
-    table: string 
-        mean, stack, or detection
-    release: string 
-        dr1 or dr2
-    baseurl: string 
-        base URL for the request
-    
-    Returns
-    -------
-    astropy table with columns name, type, description
-    """
-    
-    checklegal(table,release)
-    url = f"{baseurl}/{release}/{table}/metadata"
-    r = requests.get(url)
-    r.raise_for_status()
-    v = r.json()
-    # convert to astropy table
-    tab = Table(rows=[(x['name'],x['type'],x['description']) for x in v],
-               names=('name','type','description'))
-    return tab
-
-
-def addfilter(dtab):
-    """Add filter name as column in detection table by translating filterID
-    
-    This modifies the table in place.  If the 'filter' column already exists,
-    the table is returned unchanged.
-    
-    Parameters
-    ----------
-    dtab: table
-        detection table
-        
-    Returns
-    -------
-    dtab: table
-    """
-    if 'filter' not in dtab.colnames:
-        # the filterID value goes from 1 to 5 for grizy
-        #id2filter = np.array(list('grizy'))
-        id2filter = np.array(['panstarrs g','panstarrs r','panstarrs i','panstarrs z','panstarrs y'])
-        dtab['filter'] = id2filter[(dtab['filterID']-1).data]
-    return dtab
-
-def improve_filter_format(tab):
-    """Add filter string to column name
-    Parameters
-    ----------
-    tab:table
-    
-    Returns
-    -------
-    tab: table
-    """
-    for filter in 'grizy':
-        col = filter+'MeanPSFMag'
-        tab[col].format = ".4f"
-        tab[col][tab[col] == -999.0] = np.nan
-            
-    return(tab)
-
-def search_lightcurve(objid):
-    """setup to pull light curve info
-    
-    Parameters
-    ----------
-    objid: string
-    
-    Returns
-    -------
-    dresults: search results
-    
-    """
-    dconstraints = {'objID': objid}
-    dcolumns = ("""objID,detectID,filterID,obsTime,ra,dec,psfFlux,psfFluxErr,psfMajorFWHM,psfMinorFWHM,
-            psfQfPerfect,apFlux,apFluxErr,infoFlag,infoFlag2,infoFlag3""").split(',')
-    # strip blanks and weed out blank and commented-out values
-    dcolumns = [x.strip() for x in dcolumns]
-    dcolumns = [x for x in dcolumns if x and not x.startswith('#')]
+import lsdb
+from dask.distributed import Client
 
 
-    #get the actual detections and light curve info for this target
-    dresults = ps1search(table='detection',release='dr2',columns=dcolumns,**dconstraints)
-    
-    return(dresults)
+from data_structures import MultiIndexDFObject
 
 
-#Do a panstarrs search
-def panstarrs_get_lightcurves(sample_table, *, radius=1/3600):
-    """Searches panstarrs for light curves from a list of input coordinates.  This is the MAIN function.
+#panstarrs light curves from hipscat catalog in S3 using lsdb
+def panstarrs_get_lightcurves(sample_table, radius):
```

Comment:
Please keep the API: radius should have a sensible default and then being an optional but mandatorily keyword argument. Or at least keep it as a mandatory keyword argument. (e.g. I want to avoid having functions that end up with a call signature of positional arguments like this `functionname(a, b, None, False, True, True)`)

Reply:
done